modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java

```diff
@@ -160,24 +160,30 @@ public static Extension getExtension(Class<?> extensionType)
     private static void parseGroupParameters(MetadataType parameterType, List<ParsedParameter> parsedParameters, ClassTypeLoader typeLoader)
     {
         stream(getType(parameterType).getDeclaredFields())
-                .filter(p -> p.isAnnotationPresent(ParameterGroup.class) || p.isAnnotationPresent(Parameter.class) || p.isAnnotationPresent(MetadataKeyPart.class))
+                .filter(MuleExtensionAnnotationParser::isParameterOrParameterContainer)
                 .forEach(field ->
                          {
-                             if (field.isAnnotationPresent(ParameterGroup.class))
+                             final Map<Class<? extends Annotation>, Annotation> annotations = toMap(field.getAnnotations());
+                             final MetadataType fieldType = typeLoader.load(field.getType());
+
+                             if (isParameterContainer(annotations.keySet(), fieldType))
                              {
                                  parseGroupParameters(getFieldMetadataType(field, typeLoader), parsedParameters, typeLoader);
                              }
                              else
                              {
-                                 ParsedParameter parsedParameter = doParseParameter(field.getName(), getFieldMetadataType(field, typeLoader),
-                                                                                    toMap(field.getAnnotations()), typeLoader.getClassLoader());
+                                 ParsedParameter parsedParameter = doParseParameter(field.getName(), fieldType, annotations, typeLoader.getClassLoader());
                                  if (parsedParameter != null)
                                  {
                                      parsedParameters.add(parsedParameter);
                                  }
                              }
                          });
+    }
 
+    private static boolean isParameterOrParameterContainer(Field paramField)
```

Comment:
this method doesn't make much sense. Why do you need to know if it's a `@Parameter`? shouldn't you just need to know if it's a `@ParameterGroup` or metadata key?